src/main/java/nl/tudelft/lifetiles/graph/models/sequence/SequenceSegment.java

```diff
@@ -154,6 +160,110 @@ public final long distanceTo(final SequenceSegment other) {
      */
     @Override
     public final int compareTo(final SequenceSegment other) {
-        return Long.compare(this.getStart(), other.getStart());
+        int candidateComp = Long.compare(this.getAbsStart(),
```

Comment:
This looks dirty. Refactoring this using a for example a `while` loop should make this more readable.